src/toil/provisioners/clusterScaler.py

```diff
@@ -235,6 +235,23 @@ def __init__(self, provisioner, jobBatcher, config):
         else:
             self.scaler = None
 
+    def check(self):
+        """
+        Attempt to join any existing scaler threads that may have died or finished. This insures
+        any exceptions raised in the threads are propagated in a timely fashion.
+        """
+        exception = False
+        for scalerThread in [self.preemptableScaler, self.scaler]:
+            if scalerThread is not None:
+                try:
+                    scalerThread.join(0)
+                except Exception as e:
+                    logger.exception(e)
+                    exception = True
+        if exception:
+            raise RuntimeError('The cluster scaler has exited due to an exception')
```

Comment:
`A scaler thread …` or `One of the …` and mention that the exception was logged.